packages/visual-editor/src/components/pageSections/ReviewsSection.tsx

```diff
@@ -98,40 +125,269 @@ const ReviewsHeader: React.FC<ReviewsHeaderProps> = (props) => {
+        style={{ overflowWrap: "anywhere" }}
+      >
+        {content}
+      </div>
+      {(isTruncated || expanded) && (
+        <ShowMoreButton expanded={expanded} setExpanded={setExpanded} />
+      )}
+    </div>
+  );
+};
+
+interface ReviewStarsProps {
+  rating: number;
+  hasDarkBackground: boolean;
+}
+
+const ReviewStarsWithRating: React.FC<ReviewStarsProps> = ({
+  rating,
+  hasDarkBackground,
+}) => {
+  return (
+    <>
+      <div className="font-bold font-body-md-fontWeight text-body-md-fontSize">
+        {rating}
+      </div>
+      <ReviewStars rating={rating} hasDarkBackground={hasDarkBackground} />
+    </>
+  );
+};
+
+const ReviewStars: React.FC<ReviewStarsProps> = (props) => {
+  const { rating, hasDarkBackground } = props;
+  const HalfStar = hasDarkBackground ? FaStarHalf : FaStarHalfAlt;
+  const starColor = hasDarkBackground
+    ? "text-white"
+    : "text-palette-primary-dark";
+  return (
+    <div className={`flex items-center gap-0.5 ${starColor}`}>
       {new Array(5)
         .fill(null)
         .map((_, i) =>
           rating - i >= 0.75 ? (
             <FaStar key={i} />
           ) : rating - i >= 0.25 ? (
-            <FaStarHalfAlt key={i} />
+            <HalfStar key={i} />
           ) : (
-            <FaRegStar key={i} />
+            <FaRegStar
+              key={i}
+              style={hasDarkBackground ? { display: "none" } : undefined}
+            />
           )
         )}
     </div>
   );
 };
 
-// function to fetch reviews from the temporary endpoint
-const fetchReviews = async () => {
+const ShowMoreButton: React.FC<{
+  expanded: boolean;
+  setExpanded: (expanded: boolean) => void;
+}> = ({ expanded, setExpanded }) => {
+  const { t } = useTranslation();
+  return (
+    <Button
+      className="font-body-fontFamily text-body-md-fontSize underline cursor-pointer inline-flex items-center"
+      onClick={() => setExpanded(!expanded)}
+      variant={"link"}
+    >
+      {expanded ? t("showLess", "Show less") : t("showMore", "Show more")}
+      <IoIosArrowDown
+        className={`ml-1 transition-transform ${expanded ? "rotate-180" : ""}`}
+        style={{ textDecoration: "underline" }}
+      />
+    </Button>
+  );
+};
+
+/**
+ * Fetch reviews for the business.
+ * @returns A promise that resolves to the reviews data fetched from the endpoint.
+ */
+async function fetchReviews() {
   const response = await fetch(TEMP_ENDPOINT);
   if (!response.ok) {
     throw new Error(`HTTP error! status: ${response.status}`);
   }
-  const data = await response.json();
-  return data;
-};
+  return response.json();
+}
+
+/**
+ * Filters the reviews to only include those that belong to the current entity.
+ * @param docs - An array of review documents.
+ * @returns An array of review documents that belong to the current entity.
+ */
+function filterReviewsToEntity(docs: any[]) {
```

Comment:
Is this temporary? I imagine you'd want to send the entity in the request instead of post filtering through a bunch of data.